safetypy/safetypy.py

```diff
@@ -492,6 +501,90 @@ def delete_response(self, responseset_id, response_id):
+        log_message = 'on GET for organisations and groups of requesting user'
+        self.log_http_status(response.status_code, log_message)
+        my_groups_and_orgs = json.loads(response.content)
+        org_id = [group['id'] for group in my_groups_and_orgs['groups'] if group['type'] == "organisation"][0]
+        return org_id
+
+    def get_all_groups_in_org(self):
+        """
+        GET all the groups in the requesting user's organisation
+        :return: all the groups of the organisation
+        """
+        response = self.authenticated_request_get(self.all_groups_url)
+        log_message = 'on GET for all groups of organisation'
+        self.log_http_status(response.status_code, log_message)
+        return response
+
+    def get_users_of_group(self, group_id):
+        """
+        GET all the users of the organisations or group
+        :param group_id: ID of organisation or group
+        :return: array of users
+        """
+        url = '{0}/{1}/users'.format(self.all_groups_url, group_id)
+        response = self.authenticated_request_get(url)
+        log_message = 'on GET for users of group: {0}'.format(group_id)
+        self.log_http_status(response.status_code, log_message)
+
+        return response.content
+
+    def add_user_to_org(self, user_data):
+        """
+        POST adds a user to organisation
+        :param user_data: data of the user to be added
+        :return: userID of the user created in the organisation
+        """
+        url = self.add_users_url
+        response = self.authenticated_request_post(url, json.dumps(user_data))
+        log_message = 'on POST for adding a user to organisation'
+        self.log_http_status(response.status_code, log_message)
+        return response.content
+
+    def add_user_to_group(self, group_id, user_data):
+        """
+        POST adds a user to organisation
+        :param user_data: contains user ID of user to be added
+        :return: userID of the user created in the organisation
+        """
+        url = '{0}/{1}/users'.format(self.all_groups_url, group_id)
+        response = self.authenticated_request_post(url, json.dumps(user_data))
+        log_message = 'on POST for adding a user to group'
+        self.log_http_status(response.status_code, log_message)
+        return response.content
+
+    def update_user(self, user_id, user_data):
+        """
+        PUT updates user details such as user status(active/inactive)
+        :param user_id: The ID of the user to update
+        :return:  None
+        """
+        url = '{0}/{1}'.format(self.add_users_url, user_id)
+        response = self.authenticated_request_put(url, json.dumps(user_data))
+        log_message = 'on PUT for updating a user'
+        self.log_http_status(response.status_code, log_message)
+        return response
+
+    def remove_user(self, role_id, user_id):
+        """
+        Removes a user from an group/organisation
```

Comment:
If every other function has GET, POST etc. should this have DELETE? My pref would be to use natural language in the others instead of changing this one though.